Khan/wonder-blocks
# WB-919 - Enable auto-dismiss of Popover component

## Summary:
Adding the ability for the `Popover` component to be auto-dismissed when
the user clicks outside of the element. This "auto-dismisse" feature
is set to only work when `dismissEnabled=true` as requested.

Issue: https://khanacademy.atlassian.net/browse/WB-919

## Test plan:
1. Open Styleguidist in the browser.
2. Go to the "Popover" section.
3. Test clicking outside the popovers that have `dismissEnabled=true`.
   You will see the popover auto-dismiss itself.

---

#### Videos of Popover being auto-dismissed
##### NOTE: Added `dismissEnabled=true` to some examples that did not have it only for testing.

#### Popover Example 1:
![popover-ad-1](https://user-images.githubusercontent.com/60367213/121266764-b4d29200-c880-11eb-81f4-ec72693f8543.gif)

#### Popover Example 2:
![popover-ad-2](https://user-images.githubusercontent.com/60367213/121266771-b603bf00-c880-11eb-9eba-928acb3ab246.gif)

#### Popover Example 3 - Dismissing when opening another popover (a big expectation from devs / users):
![popover-ad-3](https://user-images.githubusercontent.com/60367213/121266776-b734ec00-c880-11eb-8561-1bc3a745fca4.gif)
